### PR TITLE
Fixes #4

### LINKED ISSUE #4: Google Colab raises errors due to Python3.7 version
We need script to easily update system if it does not match our requirements.

### PR DESCRIPTION
**Pull request by SphericalPotatoInVacuum**

### CODE REVIEW

File: src/storage/simple.py [hunk 1]
```diff
@@ -34,7 +34,7 @@ def save(self, epoch: int, modules: dict[str, nn.Module], metric: dict[str, floa
         '''
         if epoch % self.save_freq == 0:
             epoch_path = self.save_path / str(epoch)
-            epoch_path.mkdir(parents=True, exist_ok=True)
+            epoch_path.mkdir()
```
Comment (E1eMenta) [inline src/storage/simple.py:37]: Why do we remove parents=True, exist_ok=True?

File: src/storage/simple.py [hunk 2]
```diff
@@ -22,7 +22,7 @@ def __init__(self, save_folder: Union[str, Path], save_freq: int = 1):
         self.save_path.mkdir(parents=True)
         self.save_freq = save_freq
 
-    def save(self, epoch: int, modules: dict[str, nn.Module], metric: dict[str, float]):
+    def save(self, epoch: int, modules: Dict[str, nn.Module], _metric: Dict[str, float]):
```
Comment (E1eMenta) [inline src/storage/simple.py:25]: why metric is changed to _metric?

Comment (SphericalPotatoInVacuum) [reply src/storage/simple.py:25]: Initially it was to supress the linter warning about the unused parameter, but to think about it, it's better to have a comment disabling that warning instead of the variable change, will fix.

File: src/storage/simple.py (hunk 1 above)
Comment (SphericalPotatoInVacuum) [reply src/storage/simple.py:37]: The parent directories are created in the `__init__`, but I think you might be right that I was too hasty to remove `exist_ok`. The logic behind the deletion was that there shouldn't be any folders there, but I forgot that we prompt the user for deletion if we detect that the folder already exists. Should we assume that if the user doesn't want to delete already existing folders then we can overwrite any files there, or should we exit and let the user backup data instead? If it's the latter then we can safely work without `exist_ok`, because there shouldn't be anything there in that case. Howerver, I'm not sure that the implementation complies with that logic, I will make sure it does.

File: src/logger/simple.py [hunk 3]
```diff
@@ -34,7 +35,7 @@ def log_train(self, losses: Optional[dict[str, float]] = None, images: Optional[
             for loss_name, loss_value in losses.items():
                 self.loss_buff['values'][loss_name] += [loss_value]
                 self.loss_buff['sumlast'][loss_name] += loss_value
-                self.wandb.log({f"{loss_name} loss": loss_value})
+                self.wandb.log({f"{loss_name} loss": loss_value})  # type: ignore
```
Comment (grenlayk) [inline src/logger/simple.py:38]: What does this comment mean?

File: src/metrics/accuracy.py [hunk 4]
```diff
@@ -1,17 +1,18 @@
...
 # source:
 # https://gist.github.com/weiaicunzai/2a5ae6eac6712c70bde0630f3e76b77b?permalink_comment_id=3662283#gistcomment-3662283
 class TopKAccuracy:
     def __init__(self, k: Union[int, Tuple]):
+        self.k: Tuple
         if isinstance(k, int):
             self.k = (k,)
         else:
-            self.k = k
+            self.k = tuple(k)
```
Comment (grenlayk) [inline src/metrics/accuracy.py:13]: Why do we need tuple(k) here?

File: src/storage/simple.py (hunk 1 above)
Comment (grenlayk) [reply src/storage/simple.py:37]: I suppose you're right that we don't need these params according to `init()` and program's logic.

However, I think we must always abort if the folder already exists. Program shouldn't be able to delete something itself even if it asks for it -- you can make a typo in the answer.

File: src/logger/simple.py (hunk 3 above)
Comment (SphericalPotatoInVacuum) [reply src/logger/simple.py]: We initialize `self.wandb` as `wandb.init(...)` which has the return type `Run | RunDisabled | None` and this comment was to disable mypy warnings that `None` has no attribute `log`. Fixed it by asserting that what we got isn't `None`.

File: src/metrics/accuracy.py (hunk 4 above)
Comment (SphericalPotatoInVacuum) [reply src/metrics/accuracy.py:13]: Almost, mypy didn't pick up that the only way we can get to this branch is for k to be a tuple, so it complained.

File: src/storage/simple.py (hunk 1 above)
Comment (SphericalPotatoInVacuum) [reply src/storage/simple.py:37]: Yeah nah, that's unlikely to make a typo between "y" and "n", so I think we're safe here, won't fix. And yes, our implementation does comply with what I proposed.
